compiler/sem/semtypes.nim

```diff
@@ -2010,6 +2010,12 @@ proc semTypeNode(c: PContext, n: PNode, prev: PType): PType =
         # skip 'owned' in type expressions and produce a warning
         localReport(c.config, n, reportSem rsemOwnedTypeDeprecated)
         result = semTypeExpr(c, n[1], prev)
+      elif (op.s == "sink" or op.s == "lent") and n.len == 2:
```

Comment:
This means we can't have overloaded single airity `sink` or `lent` routines?

(just an observation, I don't think it needs to change or if that's something we should even allow).

Reply:
They are generally possible, but are not considered in type contexts.

I think that this is an improvement over how it worked previously, where the presence of a `sink` routine would make all following usages of the `sink` type modifier an error (except when the `sink` routine is a single arity template with a `typedesc`/`untyped`/`typed` parameter).